### PR TITLE
As a starting point loading pretained paliGemma weights into PiZero model

### PR DESCRIPTION
**Pull request by ramkumarkoppu**
You need to have access to the gated PaliGemma model weights from the hugging-face to try this example code.

### CODE REVIEW

**Comment by ramkumarkoppu** [comment]
to address https://github.com/lucidrains/pi-zero-pytorch/issues/1

**Comment by lucidrains** [comment]
@ramkumarkoppu hey Ram, this looks like a start, but you didn't actually map the parameters of each layer to the ones here

**Comment by ramkumarkoppu** [comment]
it should be doing here:

# Map weights from PaliGemma to π₀ model. This example assumes a direct mapping
# for simplicity. You may need to customize this based on layer compatibility.
for name, param in paligemma_state_dict.items():
    if name in pi_zero_state_dict:
        pi_zero_state_dict[name].data.copy_(param.data)

**Comment by lucidrains** [comment]
I see, no worries, I'll take care of it

**Comment by ramkumarkoppu** [comment]
is this the [(https://github.com/lucidrains/pi-zero-pytorch/blob/main/pi_zero_pytorch/pi_zero.py#L592) ](https://github.com/lucidrains/pi-zero-pytorch/blob/main/pi_zero_pytorch/pi_zero.py#L592) existing method to load weights?

**Comment by ramkumarkoppu** [comment]
Hi Phil, Check the updated code in the PR, hopefully this will address the issue, let me know your thoughts.

**Comment by lucidrains** [comment]
hey Ram, that doesn't look right, as my modules aren't named one-to-one to the hf implementation

you need to identify the key weights and biases of attention, feedforward, embedding, and unembedding layers here and from hf and match them up and load them. then you need to test it by running an input and asserting that the logits are equal

that's ok if it is too difficult, i can take care of it before end of month.

**Comment by ramkumarkoppu** [comment]
Hi Phil, understood, I may make further changes and test it in my forked branch and will create a PR if it works. do you want this PR to be open or shall I close it? or do you want it to be opened while we both working in separate branches?

**Comment by lucidrains** [comment]
@ramkumarkoppu i would say close it for now, thank you for the effort Ram!

**Comment by ramkumarkoppu** [comment]
I will try to make further changes and test it in my forked branch and will create a PR if it works
